Pharmacy/PharamcyApi/Controllers/MedicineController.cs

```diff
@@ -6,24 +6,29 @@
 using Pharmacy.Model;
 using Pharmacy.Repositories;
 using Pharmacy.Repositories.Base;
+using PharmacyApi.ConfigurationMappers;
+using PharmacyApi.Controllers.Base;
 using PharmacyApi.DTO;
 
 namespace PharmacyApi.Controllers
 {
     [Route("api/[controller]/[action]")]
     [ApiController]
-    public class MedicineController : Controller
+    public class MedicineController : BasePharmacyController
     {
-        private readonly IUnitOfWork _uow;
 
-        public MedicineController(IUnitOfWork uow)
+        public MedicineController(IUnitOfWork uow, PharmacyDetails details) : base(uow, details)
         {
-            _uow = uow;
         }
 
         [HttpPost]
         public IActionResult Create(CreateMedicineDTO medicineDTO)
         {
+
+            if (!IsApiKeyValid(medicineDTO.ApiKey))
+                return BadRequest(ModelState);
```

Comment:
We should say what was wrong. Like this we are just returning BadRequest but we don't state what the error was. ModelState has a method AddModelError which can be used to specify a new error and link it with a field that it refers to.

Reply:
My mistake, we had it implemented in the base controller